src/app/service/ndkprovider.service.ts

```diff
@@ -271,15 +268,22 @@ export class NdkproviderService {
     return user?.profile;
   }
 
-  async fetchFollowersFromNpub(npub: string):Promise<Set<NDKUser> | undefined>{
+  async fetchFollowersFromNpub(npub: string): Promise<Set<NDKUser> | undefined> {
     const user = this.ndk?.getUser({ npub });
     const ndkUsers = await user?.follows();
     return ndkUsers;
   }
 
   async getNdkUserFromNpub(npub: string): Promise<NDKUser | undefined> {
     try {
-      const user = this.ndk?.getUser({ npub });
+      let relayUrls: string[] | undefined = [];
+      let user: NDKUser | undefined;
+      if (localStorage.getItem(Constants.RELAYSUBS) !== undefined){
+        relayUrls = localStorage.getItem(Constants.RELAYSUBS)?.split(',');
+        user = this.ndk?.getUser({npub: npub, relayUrls: relayUrls})
```

Comment:
Here, the `getUser()` function takes an object of type `GetUserParam` as input.  This doesn't seem to have `relayUrls` in its definition

Here is what I see in the type def

```js
interface GetUserParams extends NDKUserParams {
    npub?: string;
    hexpubkey?: string;
}
```
Hence, this if condition can be removed and we could use the same getUser call for all cases.

Reply:
The NDKUserParams interface has a relayUrls property. Since GetUserParams extends NDKUserParams, it should be available. If you don't like it, we can get rid of it. I just didn't see another way to assign relays to an NDKUser and I noticed a few `this.currentUser?.relayUrls` checks. 

```

export interface NDKUserParams {
    npub?: string;
    hexpubkey?: string;
    nip05?: string;
    relayUrls?: string[];
}

```

Reply:
I just took care of it

Reply:
> The NDKUserParams interface has a relayUrls property. Since GetUserParams extends NDKUserParams, it should be available. If you don't like it, we can get rid of it. I just didn't see another way to assign relays to an NDKUser and I noticed a few `this.currentUser?.relayUrls` checks.
> 
> ```
> 
> export interface NDKUserParams {
>     npub?: string;
>     hexpubkey?: string;
>     nip05?: string;
>     relayUrls?: string[];
> }
> ```

My bad.. You are right. i will reinstate your earlier code. 

My takeaway: Doing early morning groggy reviews = bad 😵‍💫